exec/src/coqbool.ml

```diff
@@ -0,0 +1,35 @@
+(* FreeSpec
+ * Copyright (C) 2018–2019 ANSSI
+ *
+ * Contributors:
+ * 2019 Thomas Letan <thomas.letan@ssi.gouv.fr>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *)
+
+open Query
+open Utils
+open Constr
+
+let bool_of_coqbool cbool =
+  match kind cbool with
+  | Construct (c, _) ->
+     (match Ind.Bool.constructor_of c with
+      | Some b -> b
+      | _ -> raise (UnsupportedTerm "not a [bool] constructor"))
+  | _ -> raise (UnsupportedTerm "not a constructor")
+
```

Comment:
The shape of this function looks like a general pattern we should probably make explicit in a general purpose module.

Reply:
Maybe we should integrate it inside the `constructor_of` then. I will probably do it, but in another PR, I think.

Reply:
See #11